{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "I8CoX4R7VoCd"
   },
   "source": [
    "# 反爬：代理伺服器/IP\n",
    "\n",
    "* 了解「IP 黑/白名單」的反爬蟲機制\n",
    "* 「IP 黑/白名單」反爬蟲的因應策略"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "colab_type": "text",
    "id": "n-3DZ_r5VoCe"
   },
   "source": [
    "## 作業目標\n",
    "\n",
    "* 目前程式中的 proxy_ips 是手動輸入的，請根據 https://free-proxy-list.net/ 寫一個可自動化抓取可用 Proxy 的 proxy_ips。\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import random\n",
    "import time\n",
    "import json\n",
    "\n",
    "requests.adapters.DEFAULT_RETRIES = 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "爬取 proxy list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "iny-9heBVoCf"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "300\n",
      "['96.9.77.203:55667', '167.71.5.83:8080', '188.226.141.61:3128', '113.53.83.212:44664', '103.208.200.114:23500', '119.28.65.21:80', '188.226.141.211:8080', '208.80.28.208:8080', '203.202.245.62:80', '154.16.63.16:3128', '5.252.161.48:3128', '154.16.202.22:3128', '162.243.108.129:8080', '191.96.71.118:3128', '41.32.229.249:80', '180.250.12.10:80', '46.4.96.137:3128', '35.235.115.241:80', '200.41.186.100:8080', '45.232.152.194:47622', '138.197.157.32:8080', '138.68.240.218:3128', '138.68.161.14:3128', '138.68.41.90:3128', '139.162.78.109:8080', '188.166.83.17:3128', '198.199.120.102:8080', '207.154.231.213:8080', '88.198.24.108:8080', '18.230.121.61:80', '135.181.36.161:8888', '163.172.28.22:80', '176.9.75.42:8080', '78.47.16.54:80', '198.199.86.11:8080', '128.199.202.122:8080', '209.97.150.167:8080', '159.203.61.169:3128', '191.96.42.80:8080', '176.9.119.170:8080', '80.48.119.28:8080', '102.129.249.120:8080', '102.165.69.79:8080', '125.163.46.24:8080', '178.215.190.240:59541', '96.9.86.70:53827', '180.246.205.208:38044', '154.0.15.166:46547', '138.68.60.8:8080', '173.212.202.65:80', '176.98.95.105:32018', '201.204.46.10:39371', '203.80.171.147:50211', '161.35.70.249:8080', '200.73.186.234:8080', '128.199.1.186:3128', '88.198.50.103:8080', '150.136.231.227:80', '212.129.2.210:3838', '94.228.85.166:38510', '134.209.29.120:3128', '103.247.219.30:36295', '103.83.116.202:55443', '165.232.105.66:8004', '155.0.181.254:43496', '62.23.15.92:3128', '163.172.229.166:3838', '46.229.187.169:53281', '163.172.61.220:3838', '65.51.251.218:3128', '185.4.135.147:3128', '109.167.226.107:38608', '51.75.147.33:3128', '193.178.146.237:3838', '124.219.176.139:39589', '43.245.202.15:57396', '175.106.10.226:51630', '103.57.70.248:52000', '181.209.82.154:23500', '118.172.201.59:49366', '185.114.137.14:45976', '185.34.17.248:58137', '154.204.50.16:80', '41.217.219.53:31398', '5.189.133.231:80', '203.176.135.98:52234', '163.172.226.90:3838', '103.148.79.114:8080', '105.27.237.30:80', '13.92.119.142:80', '195.154.165.153:3838', '163.172.11.160:3838', '103.103.175.253:3128', '191.101.39.121:80', '212.83.178.76:3838', '51.210.106.68:443', '51.15.152.2:3838', '109.94.220.195:8085', '102.64.122.140:8085', '102.64.123.250:8085', '142.93.165.24:80', '93.177.116.50:8085', '102.64.120.238:8085', '5.135.204.120:3128', '77.81.107.91:54321', '167.172.111.6:80', '37.187.99.146:3128', '192.168.0.125:808', '192.168.0.129:808', '139.59.208.248:80', '167.172.173.32:80', '92.24.120.91:3128', '134.209.26.163:80', '45.134.187.73:80', '157.245.108.181:80', '139.59.138.119:80', '165.227.152.220:6666', '161.35.216.188:80', '139.59.133.126:80', '139.162.235.223:80', '139.59.165.135:80', '202.137.28.90:5050', '193.8.127.176:80', '139.59.12.75:80', '142.93.210.110:80', '45.168.82.28:8088', '172.104.140.243:80', '161.35.22.130:80', '5.9.139.148:8080', '139.59.214.131:80', '142.93.105.197:80', '167.172.175.36:80', '206.189.27.120:80', '209.97.177.138:8080', '3.10.236.207:80', '45.136.231.115:80', '142.93.220.190:80', '45.175.123.4:8080', '138.197.174.29:80', '139.162.158.253:80', '139.59.136.39:80', '167.71.54.123:80', '139.162.135.28:80', '78.47.151.98:8888', '88.198.203.80:80', '139.59.215.100:80', '86.57.181.122:35709', '165.227.63.57:80', '94.231.216.67:8085', '162.243.243.76:80', '54.169.132.25:80', '93.177.116.223:8085', '109.94.220.113:8085', '139.162.26.81:80', '45.137.40.189:80', '45.57.204.28:3128', '109.94.220.215:8085', '45.142.28.217:80', '165.227.188.27:80', '167.71.166.111:80', '170.79.85.224:55443', '185.23.131.240:8080', '161.35.139.194:80', '175.212.226.20:80', '193.8.127.152:80', '35.185.131.255:80', '93.177.117.15:8085', '193.8.127.56:80', '13.244.57.194:80', '45.86.15.100:80', '168.81.130.4:3128', '45.228.96.34:56707', '109.94.220.136:8085', '94.231.216.81:8085', '93.177.116.75:8085', '45.134.187.68:80', '109.236.52.28:8085', '93.177.117.136:8085', '94.231.216.32:8085', '45.142.28.79:80', '142.93.212.45:80', '139.59.143.35:80', '165.227.159.158:80', '104.248.130.32:80', '173.249.51.179:3128', '176.56.107.200:33911', '178.33.148.177:8080', '51.68.91.191:80', '139.162.255.245:80', '139.162.194.206:80', '139.162.195.127:80', '206.189.18.239:80', '206.189.114.111:8080', '209.97.134.47:80', '35.176.239.127:3128', '209.97.191.221:8080', '3.9.157.119:80', '3.8.156.108:80', '119.81.189.194:8123', '182.253.31.152:8080', '103.134.213.51:8080', '36.94.17.13:8080', '139.59.75.46:80', '196.18.207.96:3128', '159.65.122.187:80', '94.231.216.17:8085', '94.231.216.45:8085', '45.7.177.237:34234', '45.137.40.205:80', '93.177.117.254:8085', '185.126.65.16:80', '45.32.60.106:80', '68.183.16.163:80', '103.15.242.214:47424', '94.231.216.104:8085', '139.59.24.65:80', '45.124.145.34:8080', '201.49.37.124:80', '187.86.193.46:80', '196.18.204.160:3128', '23.236.182.191:80', '93.177.117.112:8085', '13.231.210.253:80', '45.57.235.84:3128', '109.94.220.127:8085', '114.5.35.98:38554', '93.177.116.202:8085', '34.211.52.170:80', '54.39.209.40:3128', '138.197.148.66:8080', '35.182.57.193:80', '68.183.206.169:80', '66.70.188.148:3128', '138.197.149.62:80', '138.197.161.57:80', '45.86.15.203:80', '45.86.15.147:80', '45.86.15.214:80', '200.41.94.121:41310', '161.35.196.32:80', '161.35.216.117:80', '165.227.146.163:80', '68.168.123.98:80', '160.116.247.108:3128', '93.177.116.228:8085', '109.94.220.66:8085', '137.135.117.5:80', '94.231.216.72:8085', '109.94.220.239:8085', '94.231.216.151:8085', '168.81.112.20:3128', '168.81.114.224:3128', '160.116.89.152:3128', '167.71.205.24:80', '93.177.116.72:8085', '51.77.123.247:80', '89.237.32.177:37647', '164.68.113.147:80', '23.229.109.164:80', '45.142.28.77:80', '196.18.206.52:3128', '93.177.117.11:8085', '64.227.97.95:80', '149.28.147.43:80', '190.4.200.147:999', '109.236.52.108:8085', '109.236.52.150:8085', '216.58.163.118:80', '51.83.68.25:3128', '51.75.25.8:3133', '45.9.122.181:80', '51.77.230.206:3128', '51.38.243.117:8080', '171.22.145.111:80', '51.38.146.63:3128', '46.18.88.186:3128', '62.210.73.38:80', '125.25.80.39:42790', '94.231.216.125:8085', '142.93.42.218:8080', '89.237.32.65:37647', '167.99.83.52:8080', '161.35.163.87:80', '146.185.21.177:3128', '3.8.146.71:80', '212.159.25.210:3128', '217.69.6.229:3128', '45.9.122.115:80', '93.177.116.183:8085', '91.93.73.228:7070', '167.71.183.228:80', '109.236.52.224:8085', '89.223.125.26:80', '93.177.117.251:8085', '193.8.127.118:80', '160.116.68.36:3128', '178.128.147.125:80', '45.142.28.244:80', '188.166.237.212:80', '139.59.91.40:80']\n"
     ]
    }
   ],
   "source": [
    "proxy_ips = []\n",
    "response = requests.get('https://free-proxy-list.net/')\n",
    "soup = BeautifulSoup(response.text, 'html.parser')\n",
    "\n",
    "proxy_info = soup.find('tbody').find_all('tr')\n",
    "\n",
    "for proxy in proxy_info:\n",
    "    tmp = proxy.find_all('td')\n",
    "    ip = tmp[0].text+':'+tmp[1].text    \n",
    "    proxy_ips.append(ip)\n",
    "\n",
    "print(len(proxy_ips))\n",
    "print(proxy_ips)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {},
    "colab_type": "code",
    "id": "75dIsHjiVoCi"
   },
   "outputs": [],
   "source": [
    "# for i in range(10):\n",
    "#     ip = random.choice(proxy_ips)\n",
    "#     print('Use', ip)\n",
    "#     try:\n",
    "#         resp = requests.get('http://ip.filefab.com/index.php',\n",
    "#                         proxies=ip[0], timeout=10)\n",
    "#         soup = BeautifulSoup(resp.text, 'html5lib')\n",
    "#         print(soup.find('h1', id='ipd').text.strip())\n",
    "#     except:\n",
    "#         print('Fail')\n",
    "#     time.sleep(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "驗證有效的 proxy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 96.9.77.203:55667\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145BC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 167.71.5.83:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 188.226.141.61:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 113.53.83.212:44664\n",
      "113.53.83.212\n",
      "113.53.83.212\n",
      "Use 103.208.200.114:23500\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714A188>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 119.28.65.21:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 188.226.141.211:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 208.80.28.208:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714EB48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 203.202.245.62:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response')))\n",
      "\n",
      "Use 154.16.63.16:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 5.252.161.48:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 154.16.202.22:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 162.243.108.129:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 191.96.71.118:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 41.32.229.249:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n",
      "Use 180.250.12.10:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 403 Forbidden')))\n",
      "\n",
      "Use 46.4.96.137:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 35.235.115.241:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response')))\n",
      "\n",
      "Use 200.41.186.100:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 403 ')))\n",
      "\n",
      "Use 45.232.152.194:47622\n",
      "45.232.152.194\n",
      "45.232.152.194\n",
      "Use 138.197.157.32:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 138.68.240.218:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 138.68.161.14:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 138.68.41.90:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.162.78.109:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 188.166.83.17:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 198.199.120.102:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 207.154.231.213:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 88.198.24.108:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 18.230.121.61:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 500 Internal Server Error')))\n",
      "\n",
      "Use 135.181.36.161:8888\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response')))\n",
      "\n",
      "Use 163.172.28.22:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 403 Forbidden')))\n",
      "\n",
      "Use 176.9.75.42:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 78.47.16.54:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Temporarily Unavailable')))\n",
      "\n",
      "Use 198.199.86.11:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 128.199.202.122:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 209.97.150.167:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 159.203.61.169:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 191.96.42.80:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 176.9.119.170:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 80.48.119.28:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response')))\n",
      "\n",
      "Use 102.129.249.120:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 102.165.69.79:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B76D0DAC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 125.163.46.24:8080\n",
      "125.163.46.24\n",
      "125.163.46.24\n",
      "Use 178.215.190.240:59541\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A948>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 96.9.86.70:53827\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77160C08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 180.246.205.208:38044\n",
      "180.246.205.208\n",
      "180.246.205.208\n",
      "Use 154.0.15.166:46547\n",
      "154.0.15.166\n",
      "154.0.15.166\n",
      "Use 138.68.60.8:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 173.212.202.65:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 500 Internal Server Error')))\n",
      "\n",
      "Use 176.98.95.105:32018\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', ConnectionResetError(10054, '遠端主機已強制關閉一個現存的連線。', None, 10054, None)))\n",
      "\n",
      "Use 201.204.46.10:39371\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77160248>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 203.80.171.147:50211\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77153DC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 161.35.70.249:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 200.73.186.234:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by SSLError(SSLError(\"bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')])\")))\n",
      "\n",
      "Use 128.199.1.186:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 403 Forbidden')))\n",
      "\n",
      "Use 88.198.50.103:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 150.136.231.227:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145448>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 212.129.2.210:3838\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n",
      "Use 94.228.85.166:38510\n",
      "94.228.85.166\n",
      "94.228.85.166\n",
      "Use 134.209.29.120:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 103.247.219.30:36295\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A548>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 103.83.116.202:55443\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E6C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 165.232.105.66:8004\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77162048>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 155.0.181.254:43496\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E4C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 62.23.15.92:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145308>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 163.172.229.166:3838\n",
      "163.172.229.166\n",
      "163.172.229.166\n",
      "Use 46.229.187.169:53281\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714A208>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 163.172.61.220:3838\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A808>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 65.51.251.218:3128\n",
      "65.51.251.218\n",
      "65.51.251.218\n",
      "Use 185.4.135.147:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n",
      "Use 109.167.226.107:38608\n",
      "109.167.226.107\n",
      "109.167.226.107\n",
      "Use 51.75.147.33:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 193.178.146.237:3838\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n",
      "Use 124.219.176.139:39589\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77160508>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 43.245.202.15:57396\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77162D88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 175.106.10.226:51630\n",
      "175.106.10.226\n",
      "175.106.10.226\n",
      "Use 103.57.70.248:52000\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77160B08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 181.209.82.154:23500\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145248>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 118.172.201.59:49366\n",
      "118.172.201.59\n",
      "118.172.201.59\n",
      "Use 185.114.137.14:45976\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77164588>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 185.34.17.248:58137\n",
      "185.34.17.248\n",
      "185.34.17.248\n",
      "Use 154.204.50.16:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Temporarily Unavailable')))\n",
      "\n",
      "Use 41.217.219.53:31398\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response')))\n",
      "\n",
      "Use 5.189.133.231:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 500 Internal Server Error')))\n",
      "\n",
      "Use 203.176.135.98:52234\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77162808>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 163.172.226.90:3838\n",
      "163.172.226.90\n",
      "163.172.226.90\n",
      "Use 103.148.79.114:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response')))\n",
      "\n",
      "Use 105.27.237.30:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714ED88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 13.92.119.142:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77153EC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 195.154.165.153:3838\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n",
      "Use 163.172.11.160:3838\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145408>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 103.103.175.253:3128\n",
      "103.103.175.253\n",
      "103.103.175.253\n",
      "Use 191.101.39.121:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Temporarily Unavailable')))\n",
      "\n",
      "Use 212.83.178.76:3838\n",
      "212.83.178.76\n",
      "212.83.178.76\n",
      "Use 51.210.106.68:443\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E948>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 51.15.152.2:3838\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n",
      "Use 109.94.220.195:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77170F88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 102.64.122.140:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77160A88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 102.64.123.250:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B771603C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 142.93.165.24:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 93.177.116.50:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A048>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 102.64.120.238:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77170688>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 5.135.204.120:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145088>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 77.81.107.91:54321\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77153AC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 167.172.111.6:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 37.187.99.146:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77162E48>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 192.168.0.125:808\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77179388>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 192.168.0.129:808\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E0C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 139.59.208.248:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 167.172.173.32:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 92.24.120.91:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714AFC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 134.209.26.163:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 45.134.187.73:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 157.245.108.181:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.59.138.119:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 165.227.152.220:6666\n",
      "165.227.152.220\n",
      "165.227.152.220\n",
      "Use 161.35.216.188:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.59.133.126:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 405 Method Not Allowed')))\n",
      "\n",
      "Use 139.162.235.223:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.59.165.135:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 202.137.28.90:5050\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145188>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 193.8.127.176:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 139.59.12.75:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 142.93.210.110:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 45.168.82.28:8088\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A948>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 172.104.140.243:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 161.35.22.130:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 5.9.139.148:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77170A88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 139.59.214.131:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 142.93.105.197:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 167.172.175.36:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 500 Internal Server Error')))\n",
      "\n",
      "Use 206.189.27.120:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 500 Internal Server Error')))\n",
      "\n",
      "Use 209.97.177.138:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77170F88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 3.10.236.207:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 45.136.231.115:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 142.93.220.190:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 45.175.123.4:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715AD48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 138.197.174.29:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.162.158.253:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.59.136.39:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 167.71.54.123:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.162.135.28:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 78.47.151.98:8888\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145688>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 88.198.203.80:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.59.215.100:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 86.57.181.122:35709\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77162B88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 165.227.63.57:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 94.231.216.67:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E388>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 162.243.243.76:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 54.169.132.25:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77160948>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 93.177.116.223:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77179A88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.94.220.113:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7717E0C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 139.162.26.81:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 45.137.40.189:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 45.57.204.28:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145148>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.94.220.215:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715AA08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.142.28.217:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 165.227.188.27:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 167.71.166.111:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 170.79.85.224:55443\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77160A08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 185.23.131.240:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77153B88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 161.35.139.194:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 175.212.226.20:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 193.8.127.152:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 35.185.131.255:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 93.177.117.15:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714AEC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 193.8.127.56:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 13.244.57.194:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 45.86.15.100:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 168.81.130.4:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E308>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.228.96.34:56707\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77164A48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.94.220.136:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714EB88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 94.231.216.81:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77153088>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 93.177.116.75:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715AB88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.134.187.68:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 109.236.52.28:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145088>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 93.177.117.136:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B771797C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 94.231.216.32:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7717E508>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.142.28.79:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 142.93.212.45:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.59.143.35:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 165.227.159.158:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 104.248.130.32:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 173.249.51.179:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B771453C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 176.56.107.200:33911\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A048>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 178.33.148.177:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B771530C8>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 51.68.91.191:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n",
      "Use 139.162.255.245:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 139.162.194.206:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.162.195.127:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 206.189.18.239:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 206.189.114.111:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E388>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 209.97.134.47:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 35.176.239.127:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77164C88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 209.97.191.221:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7716D788>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 3.9.157.119:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 BAD_REQUEST')))\n",
      "\n",
      "Use 3.8.156.108:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 119.81.189.194:8123\n",
      "119.81.189.194\n",
      "119.81.189.194\n",
      "Use 182.253.31.152:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714A788>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 103.134.213.51:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E648>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 36.94.17.13:8080\n",
      "36.66.102.98\n",
      "36.94.17.13\n",
      "Use 139.59.75.46:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 196.18.207.96:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714A088>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 159.65.122.187:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 94.231.216.17:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A848>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 94.231.216.45:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B76D0D9C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.7.177.237:34234\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715FD08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.137.40.205:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 93.177.117.254:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145188>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 185.126.65.16:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 45.32.60.106:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 68.183.16.163:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 103.15.242.214:47424\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714EB88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 94.231.216.104:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77164148>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 139.59.24.65:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 45.124.145.34:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715F608>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 201.49.37.124:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 405 Method Not Allowed')))\n",
      "\n",
      "Use 187.86.193.46:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))\n",
      "\n",
      "Use 196.18.204.160:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A908>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 23.236.182.191:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 93.177.117.112:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7716DE88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 13.231.210.253:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7712CE48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.57.235.84:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715AFC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.94.220.127:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E7C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 114.5.35.98:38554\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715FF88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 93.177.116.202:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145F48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 34.211.52.170:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 54.39.209.40:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77164808>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 138.197.148.66:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77137F48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 35.182.57.193:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 68.183.206.169:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 66.70.188.148:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145BC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 138.197.149.62:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 138.197.161.57:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 45.86.15.203:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 45.86.15.147:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 45.86.15.214:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 200.41.94.121:41310\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715F088>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 161.35.196.32:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 161.35.216.117:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 165.227.146.163:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 405 Method Not Allowed')))\n",
      "\n",
      "Use 68.168.123.98:80\n",
      "68.168.123.98\n",
      "68.168.123.98\n",
      "Use 160.116.247.108:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715AF08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 93.177.116.228:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714EF48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.94.220.66:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714AFC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 137.135.117.5:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 94.231.216.72:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77161BC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.94.220.239:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714AB88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 94.231.216.151:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715AF88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 168.81.112.20:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714ED08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 168.81.114.224:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715FC08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 160.116.89.152:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77164F08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 167.71.205.24:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 93.177.116.72:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145DC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 51.77.123.247:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77161E08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 89.237.32.177:37647\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7712FF88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 164.68.113.147:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145048>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 23.229.109.164:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 45.142.28.77:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 196.18.206.52:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715FA88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 93.177.117.11:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714EA88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 64.227.97.95:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 149.28.147.43:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 190.4.200.147:999\n",
      "190.4.200.147\n",
      "190.4.200.147\n",
      "Use 109.236.52.108:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E0C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.236.52.150:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715AC88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 216.58.163.118:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7716E608>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 51.83.68.25:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715A948>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 51.75.25.8:3133\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77164908>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 45.9.122.181:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 51.77.230.206:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E188>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 51.38.243.117:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715F5C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 171.22.145.111:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 51.38.146.63:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715F1C8>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 46.18.88.186:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145D08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 62.210.73.38:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7716EB48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 125.25.80.39:42790\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B76D0DA48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 94.231.216.125:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B771459C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 142.93.42.218:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7715FA88>: Failed to establish a new connection: [WinError 10061] 無法連線，因為目標電腦拒絕連線。')))\n",
      "\n",
      "Use 89.237.32.65:37647\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', ConnectionResetError(10054, '遠端主機已強制關閉一個現存的連線。', None, 10054, None)))\n",
      "\n",
      "Use 167.99.83.52:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E088>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 161.35.163.87:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 146.185.21.177:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 3.8.146.71:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 401 Not Authorized')))\n",
      "\n",
      "Use 212.159.25.210:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E288>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 217.69.6.229:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7716EBC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.9.122.115:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 93.177.116.183:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714AF48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 91.93.73.228:7070\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', ConnectionResetError(10054, '遠端主機已強制關閉一個現存的連線。', None, 10054, None)))\n",
      "\n",
      "Use 167.71.183.228:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 404 Not Found')))\n",
      "\n",
      "Use 109.236.52.224:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77164988>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 89.223.125.26:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 93.177.117.251:8085\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714E4C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 193.8.127.118:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 160.116.68.36:3128\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7714EE88>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 178.128.147.125:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 45.142.28.244:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required')))\n",
      "\n",
      "Use 188.166.237.212:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n",
      "Use 139.59.91.40:80\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 400 Bad Request')))\n",
      "\n"
     ]
    }
   ],
   "source": [
    "proxy_list = []\n",
    "\n",
    "for ip in proxy_ips:\n",
    "    print('Use', ip)\n",
    "    try:\n",
    "        resp = requests.get('https://httpbin.org/ip', proxies={'http': ip, 'https': ip},  timeout=30, \n",
    "                            headers = {'Connection': 'close'})\n",
    "        origin = json.loads(resp.text)['origin']\n",
    "        print(origin)\n",
    "        address = ip.split(':')[0]\n",
    "        print(address)\n",
    "        if address == origin:\n",
    "            proxy_list.append(ip)\n",
    "    except Exception as e:\n",
    "        print(e)\n",
    "        print()\n",
    "        pass\n",
    "    time.sleep(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['113.53.83.212:44664', '45.232.152.194:47622', '125.163.46.24:8080', '180.246.205.208:38044', '154.0.15.166:46547', '94.228.85.166:38510', '163.172.229.166:3838', '65.51.251.218:3128', '109.167.226.107:38608', '175.106.10.226:51630', '118.172.201.59:49366', '185.34.17.248:58137', '163.172.226.90:3838', '103.103.175.253:3128', '212.83.178.76:3838', '165.227.152.220:6666', '119.81.189.194:8123', '68.168.123.98:80', '190.4.200.147:999']\n"
     ]
    }
   ],
   "source": [
    "print(proxy_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "再次驗證 proxy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 190.4.200.147:999\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7716E048>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 119.81.189.194:8123\n",
      "Your IP address: 119.81.189.194\n",
      "Use 154.0.15.166:46547\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response')))\n",
      "\n",
      "Use 163.172.229.166:3838\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Unavailable')))\n",
      "\n",
      "Use 103.103.175.253:3128\n",
      "Your IP address: 103.103.175.253\n",
      "Use 163.172.229.166:3838\n",
      "Your IP address: 163.172.229.166\n",
      "Use 163.172.229.166:3838\n",
      "Your IP address: 163.172.229.166\n",
      "Use 94.228.85.166:38510\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B7716E208>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 163.172.229.166:3838\n",
      "Your IP address: 163.172.229.166\n",
      "Use 165.227.152.220:6666\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', ConnectionResetError(10054, '遠端主機已強制關閉一個現存的連線。', None, 10054, None)))\n",
      "\n",
      "Use 103.103.175.253:3128\n",
      "Your IP address: 103.103.175.253\n",
      "Use 118.172.201.59:49366\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B77145208>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 45.232.152.194:47622\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', ConnectionResetError(10054, '遠端主機已強制關閉一個現存的連線。', None, 10054, None)))\n",
      "\n",
      "Use 65.51.251.218:3128\n",
      "Your IP address: 65.51.251.218\n",
      "Use 125.163.46.24:8080\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', RemoteDisconnected('Remote end closed connection without response')))\n",
      "\n",
      "Use 163.172.229.166:3838\n",
      "Your IP address: 163.172.229.166\n",
      "Use 113.53.83.212:44664\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B76D0DEC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 103.103.175.253:3128\n",
      "Your IP address: 103.103.175.253\n",
      "Use 119.81.189.194:8123\n",
      "Your IP address: 119.81.189.194\n",
      "Use 190.4.200.147:999\n",
      "HTTPSConnectionPool(host='httpbin.org', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000023B76C19EC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(20):\n",
    "    ip = random.choice(proxy_list)\n",
    "    print('Use', ip)\n",
    "    try:\n",
    "        resp = requests.get('https://httpbin.org/ip',\n",
    "                        proxies={'http': ip, 'https': ip}, timeout=30, headers = {'Connection': 'close'})\n",
    "        origin = json.loads(resp.text)['origin']\n",
    "        print('Your IP address:', origin)\n",
    "    except Exception as e:\n",
    "        print(e)\n",
    "        print()\n",
    "    time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use 212.83.178.76:3838\n",
      "Your IP address: 212.83.178.76\n",
      "Use 65.51.251.218:3128\n",
      "Your IP address: 65.51.251.218\n",
      "Use 68.168.123.98:80\n",
      "HTTPConnectionPool(host='68.168.123.98', port=80): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B771A6F08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 212.83.178.76:3838\n",
      "HTTPConnectionPool(host='212.83.178.76', port=3838): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ReadTimeoutError(\"HTTPConnectionPool(host='212.83.178.76', port=3838): Read timed out. (read timeout=30)\"))\n",
      "\n",
      "Use 180.246.205.208:38044\n",
      "HTTPConnectionPool(host='180.246.205.208', port=38044): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B771AE8C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 103.103.175.253:3128\n",
      "Your IP address: 103.103.175.253\n",
      "Use 212.83.178.76:3838\n",
      "Your IP address: 212.83.178.76\n",
      "Use 125.163.46.24:8080\n",
      "HTTPConnectionPool(host='125.163.46.24', port=8080): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B771F9388>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 212.83.178.76:3838\n",
      "Your IP address: 212.83.178.76\n",
      "Use 113.53.83.212:44664\n",
      "Your IP address: 113.53.83.212\n",
      "Use 180.246.205.208:38044\n",
      "HTTPConnectionPool(host='180.246.205.208', port=38044): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B77241E48>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 165.227.152.220:6666\n",
      "HTTPConnectionPool(host='165.227.152.220', port=6666): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B77247F08>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.167.226.107:38608\n",
      "HTTPConnectionPool(host='109.167.226.107', port=38608): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B7724DFC8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 109.167.226.107:38608\n",
      "HTTPConnectionPool(host='109.167.226.107', port=38608): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B7725A0C8>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 180.246.205.208:38044\n",
      "HTTPConnectionPool(host='180.246.205.208', port=38044): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B7725F188>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 163.172.229.166:3838\n",
      "HTTPConnectionPool(host='163.172.229.166', port=3838): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ReadTimeoutError(\"HTTPConnectionPool(host='163.172.229.166', port=3838): Read timed out. (read timeout=30)\"))\n",
      "\n",
      "Use 113.53.83.212:44664\n",
      "HTTPConnectionPool(host='113.53.83.212', port=44664): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000023B77252648>: Failed to establish a new connection: [WinError 10060] 連線嘗試失敗，因為連線對象有一段時間並未正確回應，或是連線建立失敗，因為連線的主機無法回應。')))\n",
      "\n",
      "Use 68.168.123.98:80\n",
      "Your IP address: 68.168.123.98\n",
      "Use 212.83.178.76:3838\n",
      "HTTPConnectionPool(host='212.83.178.76', port=3838): Max retries exceeded with url: http://ip.filefab.com/index.php (Caused by ReadTimeoutError(\"HTTPConnectionPool(host='212.83.178.76', port=3838): Read timed out. (read timeout=30)\"))\n",
      "\n",
      "Use 163.172.226.90:3838\n",
      "Your IP address: 163.172.226.90\n"
     ]
    }
   ],
   "source": [
    "for i in range(20):\n",
    "    ip = random.choice(proxy_list)\n",
    "    print('Use', ip)\n",
    "    try:\n",
    "        resp = requests.get('http://ip.filefab.com/index.php',\n",
    "                        proxies={'http': ip, 'https': ip}, timeout=30, headers = {'Connection': 'close'})\n",
    "        soup = BeautifulSoup(resp.text, 'html5lib')\n",
    "        print(soup.find('h1', id='ipd').text.strip())\n",
    "    except Exception as e:\n",
    "        print(e)\n",
    "        print()\n",
    "    time.sleep(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "colab": {
   "name": "homework.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
